packages/flutter/lib/src/cupertino/switch.dart

```diff
@@ -216,52 +214,42 @@ class _CupertinoSwitchState extends State<CupertinoSwitch> with TickerProviderSt
   }
 
   void _handleTapDown(TapDownDetails details) {
-    if (isInteractive)
-      needsPositionAnimation = false;
-      _reactionController.forward();
+    needsPositionAnimation = false;
+    _reactionController.forward();
   }
 
   void _handleTap() {
-    if (isInteractive) {
-      widget.onChanged!(!widget.value);
-      _emitVibration();
-    }
+    widget.onChanged!(!widget.value);
```

Comment:
```suggestion
    widget.onChanged?.call(!widget.value);
```

Reply:
Otherwise when the widget rebuilds with the `onChanged` callback set to null it may crash?